partner_validate_vat/models/res_partner.py

```diff
@@ -0,0 +1,35 @@
+# coding: utf-8
+# Copyright 2017 Vauxoo (https://www.vauxoo.com) <info@vauxoo.com>
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from openerp import models, api, fields, _
+from openerp.exceptions import ValidationError
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    vat = fields.Char(copy=False)
+
+    @api.constrains('vat', 'country_id', 'commercial_partner_id')
+    def _check_vat_uniqueness(self):
+        """Check that the vat is unique in the level where the partner is
+        in the tree
+        """
+
+        user_company = self.env.user.company_id
+
+        for partner in self.filtered(
+                lambda r: r.country_id == user_company.country_id and
```

Comment:
Add in the docstring the rationale of use `r.country_id == user_company.country_id`
Please, don't add the `what`, add the `why`